packages/contracts-bedrock/scripts/checks/opcm-upgrade-checks/main.go

```diff
@@ -0,0 +1,227 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/ethereum-optimism/optimism/op-chain-ops/solc"
+	"github.com/ethereum-optimism/optimism/packages/contracts-bedrock/scripts/checks/common"
+)
+
+var opcmArtifactPath = "forge-artifacts/OPContractsManager.sol/OPContractsManagerUpgrader.json"
+var opcmAst *solc.ForgeArtifact
+var opcmUpgradeFunctionSelector = "ff2dd5a1"
+
+func main() {
+	var err error
+
+	// Get OPCM's AST.
+	opcmAst, err = common.ReadForgeArtifact(opcmArtifactPath)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	// Process.
+	if _, err := common.ProcessFilesGlob(
+		[]string{"forge-artifacts/**/*.json"},
+		[]string{"forge-artifacts/OPContractsManager.sol/*.json"},
+		processFile,
+	); err != nil {
+		fmt.Printf("error: %v\n", err)
+		os.Exit(1)
+	}
+}
+
+func processFile(artifactPath string) (*common.Void, []error) {
+	// Get the artifact.
+	artifact, err := common.ReadForgeArtifact(artifactPath)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	// If the absolute path is not src/L1, return early.
+	if !strings.HasPrefix(artifact.Ast.AbsolutePath, "src/L1") {
+		return nil, nil
+	}
+
+	// Find if it contains any upgrade function
+	numOfUpgradeFunctions := getNumberOfUpgradeFunctions(artifact)
+
+	// If there are no upgrade functions, return early.
+	if numOfUpgradeFunctions == 0 {
+		return nil, nil
+	}
+
+	// If there are more than 1 upgrade functions, return an error.
+	if numOfUpgradeFunctions > 1 {
+		return nil, []error{fmt.Errorf("expected 0 or 1 upgrade function, found %v", numOfUpgradeFunctions)}
+	}
+
```

Comment:
Nit, if you want to simplify a bit you can just load opcmAst here, not a huge benefit to doing it globally